Intersect.Server/Entities/Npc.cs

```diff
@@ -1188,31 +1200,34 @@ private void ResetAggroCenter(out Guid targetMap)
             targetMap = Guid.Empty;
 
             // Reset our aggro center so we can get "pulled" again.
-            AggroCenterMap = null;
-            AggroCenterX = 0;
-            AggroCenterY = 0;
-            AggroCenterZ = 0;
-            mPathFinder?.SetTarget(null);
-            mResetting = false;
+            _aggroCenterMap = null;
+            _aggroCenterX = 0;
+            _aggroCenterY = 0;
+            _aggroCenterZ = 0;
+            _pathFinder?.SetTarget(null);
+            _resetting = false;
         }
 
         private bool CheckForResetLocation(bool forceDistance = false)
         {
             // Check if we've moved out of our range we're allowed to move from after being "aggro'd" by something.
             // If so, remove target and move back to the origin point.
-            if (Options.Npc.AllowResetRadius && AggroCenterMap != null && (GetDistanceTo(AggroCenterMap, AggroCenterX, AggroCenterY) > Math.Max(Options.Npc.ResetRadius, Math.Min(Base.ResetRadius, Math.Max(Options.MapWidth, Options.MapHeight))) || forceDistance))
+            if (!Options.Npc.AllowResetRadius || _aggroCenterMap == null ||
+                (GetDistanceTo(_aggroCenterMap, _aggroCenterX, _aggroCenterY) <= Math.Max(Options.Npc.ResetRadius,
+                    Math.Min(Base.ResetRadius, Math.Max(Options.MapWidth, Options.MapHeight))) && !forceDistance))
```

Comment:
Can you think of a way to refactor this?

Reply:
```csharp
        private bool ShouldResetLocation(bool forceDistance)
        {
            if (!Options.Npc.AllowResetRadius || _aggroCenterMap == null)
            {
                return false;
            }

            var resetRadius = Math.Max(Options.Npc.ResetRadius, Math.Min(Base.ResetRadius, Math.Max(Options.MapWidth, Options.MapHeight)));
            var distanceToAggroCenter = GetDistanceTo(_aggroCenterMap, _aggroCenterX, _aggroCenterY);
            bool withinResetRadius = distanceToAggroCenter <= resetRadius && !forceDistance;

            return !withinResetRadius;
        }
```

Reply:
Great!